tests/ecosystem/upgrade/conftest.py

```diff
@@ -0,0 +1,309 @@
+import logging
+
+from ocs_ci.ocs import constants, resources
+import pytest
+import threading
+
+log = logging.getLogger(__name__)
+
+
+def create_pods(
+    interface,
+    pvc_factory,
+    pod_factory,
+    storageclass,
+    count,
+    access_mode,
+    volume_mode=None
+):
+    """
+    Create pods for upgrade testing. pvc_factory and pod_factory have to be
+    in the same scope.
+
+    Args:
+        interface (str): CephBlockPool or CephFileSystem
+        pvc_factory (function): Function for creating PVCs
+        pod_factory (function): Function for creating pods
+        storageclass (obj): Storageclass to use
+        count (int): Number of pods to create
+        access_mode (str): ReadWriteOnce, ReadOnlyMany or ReadWriteMany.
+            This decides the access mode to be used for the PVC
+        volume_mode (str): Volume mode for rbd RWO PVC
+
+    Return:
+        list: List of generated pods
+    """
+    log.info(
+        f"Creating {count} pods via {interface} using {access_mode}"
+        f" access mode, {volume_mode} volume mode and {storageclass.name}"
+        f" storageclass"
+    )
+    pvcs = [
+        pvc_factory(
+            storageclass=storageclass,
+            access_mode=access_mode,
+            volume_mode=volume_mode
+        ) for _ in range(count)
+    ]
+    if volume_mode == constants.VOLUME_MODE_BLOCK:
+        pod_dict = constants.CSI_RBD_RAW_BLOCK_POD_YAML
+        raw_block_pv = True
+    else:
+        pod_dict = ''
+        raw_block_pv = False
+    pods = [
+        pod_factory(
+            interface=interface,
+            pvc=pvc,
+            raw_block_pv=raw_block_pv,
+            pod_dict_path=pod_dict,
+        ) for pvc in pvcs
+    ]
+    return pods
+
+
+@pytest.fixture(scope='session')
+def default_storageclasses(request, teardown_factory_session):
+    """
+    Returns dictionary with storageclasses. Keys represent reclaim policy of
+    storageclass. There are two storageclasses for each key. First is RBD based
+    and the second one is CephFS based.
+    """
+    scs = {
+        constants.RECLAIM_POLICY_DELETE: [],
+        constants.RECLAIM_POLICY_RETAIN: []
+    }
+
+    # TODO(fbalak): Use proper constants after
+    # https://github.com/red-hat-storage/ocs-ci/issues/1056
+    # is resolved
+    for sc_name in (
+        'ocs-storagecluster-ceph-rbd',
+        'ocs-storagecluster-cephfs'
+    ):
+        sc = resources.ocs.OCS(
+            kind=constants.STORAGECLASS,
+            metadata={'name': sc_name}
+        )
+        sc.reload()
+        scs[constants.RECLAIM_POLICY_DELETE].append(sc)
+        sc.data['reclaimPolicy'] = constants.RECLAIM_POLICY_RETAIN
+        sc.data['metadata']['name'] += '-retain'
+        sc._name = sc.data['metadata']['name']
+        sc.create()
+        teardown_factory_session(sc)
+        scs[constants.RECLAIM_POLICY_RETAIN].append(sc)
+    return scs
+
+
+@pytest.fixture(scope='session')
+def pre_upgrade_filesystem_pods(
```

Comment:
I suggest, just provide the fixtures as is and let users use it where they want it.

Reply:
You mean to move them to tests/conftest.py?
I don't think that these fixtures should be used in other places. These are designed specially for upgrade and should be used only in upgrade test cases. 
Where do you want to use them?